arangod/Agency/Supervision.cpp

```diff
@@ -1121,32 +1132,62 @@ void Supervision::workJobs() {
   auto todos = _snapshot.hasAsChildren(toDoPrefix).first;
   auto it = todos.begin();
   static std::string const FAILED = "failed";
-    
+
+  size_t const maximalJobsPerRound = 1000;
+  bool selectRandom = todos.size() > maximalJobsPerRound;
```

Comment:
I will add these sentences first thing tomorrow morning. It is possible that jobs in ToDo are inspected and in fact remain in ToDo, because they are waiting for something. Same for pending. The randomization is needed to give every job a spin eventually. I will explain this in a comment.

Reply:
Done.